ci-operator/step-registry/openshift-observability/enable-cluster-logging/openshift-observability-enable-cluster-logging-commands.sh

```diff
@@ -0,0 +1,605 @@
+function deploy_minio()
+{
+    echo "### deploy minio for lokistack in ${MINIO_NAMESPACE}"
+    prepare_namespace ${MINIO_NAMESPACE}
+    minio_secret="minio_$(date +%s)"
+
+    cat <<EOF |oc create -f -
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: minio-pv-claim
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: minio
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  selector:
+    matchLabels:
+      app: minio
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: minio
+    spec:
+      volumes:
+      - name: data
+        persistentVolumeClaim:
+          claimName: minio-pv-claim
+      containers:
+      - name: minio
+        volumeMounts:
+        - name: data
+          mountPath: "/data"
+        image: quay.io/openshifttest/minio:latest
+        args:
+        - server
+        - /data
+        - --console-address
+        - ":9001"
+        env:
+        - name: MINIO_ACCESS_KEY
+          value: "minio"
+        - name: MINIO_SECRET_KEY
+          value: "${minio_secret}"
+        ports:
+        - containerPort: 9000
+        readinessProbe:
+          httpGet:
+            path: /minio/health/ready
+            port: 9000
+          initialDelaySeconds: 120
+          periodSeconds: 20
+        livenessProbe:
+          httpGet:
+            path: /minio/health/live
+            port: 9000
+          initialDelaySeconds: 120
+          periodSeconds: 20
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: minio-service
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  ports:
+    - port: 9000
+      targetPort: 9000
+      protocol: TCP
+  selector:
+    app: minio
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: minio-service-console
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  ports:
+    - port: 9001
+      targetPort: 9001
+      protocol: TCP
+  selector:
+    app: minio
+---
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  labels:
+    app: minio
+  name: minio
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  port:
+    targetPort: 9000
+  to:
+    kind: Service
+    name: minio-service
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+---
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  labels:
+    app: minio
+  name: minio-console
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  port:
+    targetPort: 9001
+  to:
+    kind: Service
+    name: minio-service-console
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+EOF
+}
+
+function wait_minio_ready(){
+    echo "### wait the mino ready up to 2 minutes"
+
+    minio_ready=false
+    count=1
+    while [[ $count -lt 8 ]]; do
+        if oc -n ${MINIO_NAMESPACE} wait pod --for=condition=Ready  -l app=minio; then
+            minio_ready=true
+        fi
+        if [[ $minio_ready == "true" ]]; then
```

Comment:
These 2 lines can be removed.

Reply:
We need this two lines to exit the script.

Reply:
Do you mean oc wait --timeout=5m?

Reply:
If we run wait pod immediately.  the command will exist with error 'no resource found', so we still need the retry step.